src/InstrumentationEngine.Lib/Logging.h

```diff
@@ -52,6 +52,9 @@ namespace MicrosoftInstrumentationEngine
         static CSingleton<CLoggerService> s_loggerService;
         static std::atomic_size_t s_initCount;
 
+        // This caches the effective loggingflag of s_loggerService for performance
+        static std::atomic<LoggingFlags> s_cachedFlags;
```

Comment:
Is this really that much more performant? What makes GetLoggingFlags so slow given that it just reads m_effectiveFlags?

Reply:
Currently, in Code coverage scenarios (which are very heavy), calls to CLogging::LogMessage take up about 8% of cpu time. If we remove the calls to initialize the logger service, and to get the logging flags, we may be able to get that down to about 4%.